pkg/velero/backup.go

```diff
@@ -0,0 +1,353 @@
+package velero
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	veleroClient "github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BackupBuilder provides a struct for backup object from the cluster and a backup definition.
+type BackupBuilder struct {
+	// Backup definition, used to create the backup object.
+	Definition *velerov1.Backup
+	// Created backup object.
+	Object *velerov1.Backup
+	// Used to store latest error message upon defining or mutating backup definition.
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient veleroClient.Interface
+}
+
+// NewBackupBuilder creates a new instance of BackupBuilder.
+func NewBackupBuilder(apiClient veleroClient.Interface, name, nsname string) *BackupBuilder {
+	glog.V(100).Infof(
+		"Initializing new backup structure with the following params: "+
+			"name: %s, namespace: %s", name, nsname)
+
+	builder := &BackupBuilder{
+		apiClient: apiClient,
+		Definition: &velerov1.Backup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the backup is empty")
+
+		builder.errorMsg = "backup name cannot be an empty string"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the backup is empty")
+
+		builder.errorMsg = "backup namespace cannot be an empty string"
+	}
+
+	return builder
+}
+
+// PullBackup loads an existing backup into BackupBuilder struct.
+func PullBackup(apiClient *clients.Settings, name, nsname string) (*BackupBuilder, error) {
```

Comment:
Could this match the change from https://github.com/openshift-kni/eco-goinfra/pull/272?

Instead of passing in the entire `clients.Settings` pointer, just pass the veleroClient interface like in`NewBackupBuilder`.